caikit/runtime/http_server/utils.py

```diff
@@ -16,31 +16,42 @@
 this includes things like parameter handles and openapi spec generation
 """
 # Standard
-from typing import Any, Optional
+from typing import Any, Dict, Optional
 
 # Local
 from ...config.config import merge_configs
 
 
-def convert_json_schema_to_multipart(json_schema):
+def convert_json_schema_to_multipart(json_schema, defs):
     """Helper function to convert a json schema from applicaiton/json into one
     that can be used for multipart requests"""
-    sparse_schema, extracted_files = _extract_raw_from_schema(json_schema)
+    sparse_schema, extracted_files = _extract_raw_from_schema(json_schema, defs)
     sparse_schema["properties"] = {
         **sparse_schema.get("properties", {}),
         **extracted_files,
     }
     return sparse_schema
 
 
-def _extract_raw_from_schema(json_schema: Any, current_path=None) -> (dict, dict):
+def _extract_raw_from_schema(
+    json_schema: Any, defs: Dict[str, Any], current_path=None
+) -> (dict, dict):
     """Helper function to extract all "bytes" or File fields from a json schema and return the
     cleaned schema dict and a dict of extracted schemas where the key is the original raw's path"""
     if isinstance(json_schema, dict):
         # If this json_schema represents a raw field extract it
         if raw_json_schema := _parse_raw_json_schema(json_schema):
             return None, {_clean_schema_path(current_path): raw_json_schema}
 
+        # If this json_schema is just a ref then just recurse on the ref
+        if "$ref" in json_schema:
```

Comment:
Im not really following what's happening here, but I think it looks correct ish?